lib/inventory/inventory_api.rb

```diff
@@ -348,6 +545,41 @@ def to_h
     end
   end
 
+  class MetricType
+    # @return [String] Full path of the metric (type)
+    attr_reader :path
+    # @return [String] Name of the metric
+    attr_reader :name
+    attr_reader :id
+    attr_reader :feed
+    attr_reader :unit
+    # @return [Long] collection interval in seconds
+    attr_reader :collection_interval
+
+    def initialize(type_hash)
+      @id = type_hash['id']
```

Comment:
Yes, because it fails with a symbol. I think this is because he input is a hash that was obtained from the remote and then translated to a hash via `JSON.parse(res)`